src/modules/dashboard/sagas/actions.ts

```diff
@@ -735,6 +739,210 @@ function* createDomainAction({
   return null;
 }
 
+function* editColonyAction({
+  payload: {
+    colonyAddress,
+    colonyName,
+    colonyDisplayName,
+    colonyAvatarImage,
+    colonyTokens = [],
+    annotationMessage,
+  },
+  meta: { id: metaId, history },
+  meta,
+}: Action<ActionTypes.COLONY_ACTION_EDIT_COLONY>) {
+  let txChannel;
+  try {
+    const apolloClient = TEMP_getContext(ContextModule.ApolloClient);
+
+    /*
+     * Validate the required values for the payment
+     */
+    if (!colonyDisplayName && colonyDisplayName !== null) {
+      throw new Error('A colony name is required in order to edit the colony');
```

Comment:
why !colonyDisplayName && colonyDisplayName !== null instead of just !colonyDisplayName ??

Reply:
Because if it's undefined I don't want to send the date, but if it's null, I want to (this means the colony display name was removed by the user)

Same goes for the avatar